compiler/modules/nback/basic.bal

```diff
@@ -45,8 +45,22 @@ function buildPrologue(llvm:Builder builder, Scaffold scaffold, bir:Position pos
 function buildBasicBlock(llvm:Builder builder, Scaffold scaffold, bir:BasicBlock block) returns BuildError? {
     scaffold.setBasicBlock(block);
     builder.positionAtEnd(scaffold.basicBlock(block.label));
+    boolean narrowRegsInited = false;
```

Comment:
It feels ugly to me to have this variable here and the rest of the related state in scaffold. Can we put narrowRegsInited into scaffold and then have a startBasicBlock function on scaffold?

Reply:
Renamed to `typeMergeFinished`, I think now it belongs in this function.